lib/api/api_client.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:dio/dio.dart';
+import 'package:retrofit/http.dart';
+import 'package:survey/models/user.dart';
+
+part 'api_client.g.dart';
+
+class Apis {
+  static const String oauthLogin = '/api/v1/oauth/token';
+  static const String me = '/api/v1/me';
+}
+
+@RestApi()
+abstract class ApiClient {
+  factory ApiClient(Dio dio, {String baseUrl}) = _ApiClient;
+
+  @GET(Apis.me)
+  Future<UserResponseData> getUser();
+}
```

Comment:
Do we gonna keep this simple cause it's an example repo or we can split it into a `UserService` to contain all user APIs

Reply:
let's keep it simple in here, like I mentioned above for `/me` (get profile) I will use graphQL, this one is now being used only as a pow 🙇  there will be another `GraphqlClient` 😁

Reply:
even if we were using the same RestApi, usually, at this level we can group all together as the single data source came from the same `domain`. I will introduce another layer which is `repository`, from there, we can have `UserRepository` or `SurveyRepository`.

When you change/add an API for example path or name or structure, in that case, we don't have to create a new Client but create a new Repository makes more sense.